### PR TITLE
Add support for multi-label visualization

### PR DESCRIPTION
**Pull request by JSabadin**
# Multiclass and Multilabel Classification Support in Luxonis-ML

## Overview

Luxonis-ML supports both multiclass and multi-label classification datasets. Our model components, including the `ClassificationHead` and the `BCEWithLogitsLoss`, were already compatible with multi-label tasks. However, there was an issue with the visualization of predictions due to the incorrect use of **softmax** for multi-label classification.

This PR improves the visualization logic to handle both multiclass and multi-label classification. The `multi_label` flag in the visualizer dictates whether **softmax** (for multiclass tasks) or **sigmoid** (for multi-label tasks) is applied, ensuring accurate representation of class probabilities based on the classification type.

## Changes

- **`multi_label` Flag**: The `multi_label` flag controls whether the visualizer operates in multiclass or multi-label mode. Set this flag to `True` to enable multi-label classification visualization, where independent probabilities for each class are displayed using `sigmoid`. Leave it as `False` (default) for multiclass tasks that use `softmax`.

### CODE REVIEW

File: luxonis_train/attached_modules/visualizers/README.md [hunk 1]
```diff
@@ -60,12 +60,13 @@ Visualizer for bounding boxes.
...
-| `thickness`    | `int`                  | `1`           | Line thickness of the font                                                |
+| Key            | Type                   | Default value | Description                                                                      |
+| -------------- | ---------------------- | ------------- | -------------------------------------------------------------------------------- |
+| `include_plot` | `bool`                 | `True`        | Whether to include a plot of the class probabilities in the visualization        |
+| `color`        | `tuple[int, int, int]` | `(255, 0, 0)` | Color of the text                                                                |
+| `font_scale`   | `float`                | `1.0`         | Scale of the font                                                                |
+| `thickness`    | `int`                  | `1`           | Line thickness of the font                                                       |
+| `multi_label`  | `bool`                 | `False`       | Set to `True` for multi-label classification, otherwise `False` for single-label |
```
Comment (kozlov721) [inline luxonis_train/attached_modules/visualizers/README.md:69]: I think it would be better to change this to `"multilabel"` (without the dash) so it's consistent with the `params.task` in metrics.

**Review by kozlov721 — APPROVED** [review]
Suggested renaming of `multi_label` to `multilabel`, otherwise LGTM!

**Review by kozlov721 — APPROVED** [review]

File: luxonis_train/attached_modules/visualizers/classification_visualizer.py [hunk 2]
```diff
@@ -33,17 +34,28 @@ def __init__(
...
+                return str(idx)
+            return self.class_names[idx]
 
     def _generate_plot(
         self, prediction: Tensor, width: int, height: int
     ) -> Tensor:
-        pred = prediction.softmax(-1).detach().cpu().numpy()
+        if self.multilabel:
+            pred = prediction.sigmoid().detach().cpu().numpy()
+        else:
```
[the remark this answers is not in the document]
Comment (JSabadin) [reply luxonis_train/attached_modules/visualizers/classification_visualizer.py:57]: Yes, binary classification is handled correctly via `softmax `for single-label classification.

[the remark this answers is not in the document]
Comment (klemen1999) [reply luxonis_train/attached_modules/visualizers/classification_visualizer.py:57]: But with binary classification (output is 1 value) the predicted confidence is not correct. Let's say the NN predicts value=2. If we use sigmoid this will mean that it is ~88% confident (after `prediction.sigmoid()`) it is this class while with softmax whatever the value the confidence will always be 1 (after `prediction.softmax()`)

[the remark this answers is not in the document]
Comment (JSabadin) [reply luxonis_train/attached_modules/visualizers/classification_visualizer.py:57]: In our binary classification task, the output of the network is a vector of length 2 ([here](https://github.com/luxonis/luxonis-train/blob/6146da6422388ed97945d42bcd1c4625c2bc9d68/luxonis_train/nodes/heads/classification_head.py#L29)). Therefore, using softmax to normalize this vector is correct, as it will produce the proper confidence values for each class.
